contracts/ETO/ETOCommitment.sol

```diff
@@ -825,15 +834,20 @@ contract ETOCommitment is
         );
     }
 
-    function isCapExceeded(bool applyDiscounts, uint256 equityTokenInt, uint256 fixedSlotsEquityTokenInt)
+    function isCapExceeded(bool applyDiscounts, uint256 equityTokenInt, uint256 fixedSlotsEquityTokenInt, uint256 equivEurUlps)
         private
         constant
         returns (bool maxCapExceeded)
-    {
+    {   
+        // check for exceeding tokens
         maxCapExceeded = _totalTokensInt + equityTokenInt > MAX_NUMBER_OF_TOKENS;
         if (applyDiscounts && !maxCapExceeded) {
             maxCapExceeded = _totalTokensInt + equityTokenInt - _totalFixedSlotsTokensInt - fixedSlotsEquityTokenInt > MAX_NUMBER_OF_TOKENS_IN_WHITELIST;
         }
+        // check for exceeding max investment amount as defined by the constraints
+        if ( equivEurUlps + _totalEquivEurUlps > ETO_TERMS_CONSTRAINTS.MAX_INVESTMENT_AMOUNT_EUR_ULPS() ) {
```

Comment:
1. do not do this check is cap is already exceeded
2. please cache `MAX_INVESTMENT_AMOUNT_EUR_ULPS` in local storage. we are wasting 900 gas units on this call